### PR TITLE
support for chrome devtools _initiator attributes 'type', 'url' and 'lineNumber'

### PR DESCRIPTION
**Pull request by beatchristen**
Hi @deviantintegral 
I'm interested also in the non-standard attributes that chrome devtools exports since around chrome 67 or so.

All new APIs should be covered by tests.

### CODE REVIEW

File: src/Initiator.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Deviantintegral\Har;
+
+use JMS\Serializer\Annotation as Serializer;
+
+/**
+ * @see https://developers.google.com/web/tools/chrome-devtools/network/reference#requests
```
Comment (deviantintegral) [inline src/Initiator.php:8]: 👏

File: src/Initiator.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ */
+final class Initiator
+{
+    /**
+     * @var string
+     * @Serializer\Type("string")
+     *
+     * parser:
+     *   Chrome's HTML parser.
```
Comment (deviantintegral) [inline src/Initiator.php:17]: Is this truly just Chrome, or any Blink-derived browser?

File: tests/src/Unit/EntryTest.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    protected function setUp(): void
+    {
+        $this->repository = new HarFileRepository(__DIR__.'/../../fixtures');
+    }
+
+    public function testHasInitiator()
+    {
+        // Initialize a repository of HAR files, with IDs being the file names.
+        $repository =
```
Comment (deviantintegral) [inline tests/src/Unit/EntryTest.php:25]: typo here? This looks like it's double-assigning the next variable.

**Review by deviantintegral — CHANGES_REQUESTED** [review]
This looks great! Thanks for contributing.

There's one typo that needs to be fixed for sure. Otherwise, what do you think about adding a fixture that includes the initiator property? That would automatically be tested by https://github.com/deviantintegral/har/blob/master/tests/src/Unit/HarTest.php for the basic import / export case.

**Comment by deviantintegral** [comment]
Looks like I didn't have the option to run CI on forks turned on. Your next commit should trigger them.

File: src/Initiator.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+     * @return Initiator
+     */
+    public function setType(string $type): self
+    {
+        $this->type = $type;
+
+        return $this;
+    }
+
+    public function getUrl(): ?\Psr\Http\Message\UriInterface
```
Comment (beatchristen) [inline src/Initiator.php:59]: added nullability infos here.

File: src/Initiator.php (hunk 1 above)
Comment (beatchristen) [reply src/Initiator.php:8]: Thanks for providing the great base to start from!

**Comment by beatchristen** [comment]
> This looks great! Thanks for contributing.
> 
> There's one typo that needs to be fixed for sure. Otherwise, what do you think about adding a fixture that includes the initiator property? That would automatically be tested by https://github.com/deviantintegral/har/blob/master/tests/src/Unit/HarTest.php for the basic import / export case.

That's what I did at the beginning, but `removeCustomFields()` auto-fails via 

```
        $a = array_filter($a, function ($key): bool {
            return !\is_string($key) || !(0 === strpos($key, '_'));
        }, \ARRAY_FILTER_USE_KEY);
```

and putting JUST `|| '_initiator' == $key` was a bit awkward.

Potentially, refactoring the current Serializer which is hardcoded in several places like `HarTestBase`, `HarFileRepository` could be split to a `HarSpec12Serializer` with the standard format compliance and a `ChromiumBlinkDevToolsSerializer` variant with all the latest bells and whistles.

That's however a topic for another issue, I think...

**Comment by deviantintegral** [comment]
Agreed on a new issue or pull request - I think that approach is a good idea, because it could also let us support future versions of HAR should they ever happen.

**Pull request merged**